app/containers/EmojiPicker/EmojiCategory.js

```diff
@@ -44,7 +49,13 @@ export default class EmojiCategory extends React.Component {
 		this.emojis = props.emojis;
 	}
 
-	shouldComponentUpdate() {
+	shouldComponentUpdate(nextProps) {
+		const { emojis: oldEmojis } = this.props;
+
+		if (nextProps.emojis.length !== oldEmojis.length) {
```

Comment:
Why should we test for new emojis?

Reply:
Because when search is performed the EmojiCategory needs to be re-rendered. And this check updates the component only if length of search results changes.